### PR TITLE
Colmap: 3.5 -> 3.6 and fix colmapWithCuda build

### PR DESCRIPTION
**Pull request by flyfloh**
###### Motivation for this change

#99098 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/colmap/default.nix [hunk 1]
```diff
@@ -7,22 +7,15 @@ assert !cudaSupport || cudatoolkit != null;
...
   src = fetchFromGitHub {
      owner = "colmap";
      repo = "colmap";
      rev = version;
-     sha256 = "1vnb62p0y2bnga173wmjs0lnyqdjikv0fkcxjzxm8187khk2lly8";
+     sha256 = "1kfivdmhpmdxjjf30rr57y2iy7xmdpg4h8aw3qgacv8ckfpgda3n";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/colmap/default.nix:20]: Remove fetchpatch from inputs.

Comment (flyfloh) [reply pkgs/applications/science/misc/colmap/default.nix:20]: Done.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107962` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>colmap</li>
<li>colmapWithCuda</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Review by blitz — APPROVED** [review]
I've tested that `colmap` and `colmapWithCuda` build and run.

@SuperSandro2000 Let's get this merged!

**Review by tfc — APPROVED** [review]

**Pull request merged**